libafl_frida/src/coverage_rt.rs

```diff
@@ -184,6 +187,24 @@ impl CoverageRuntime {
     pub fn emit_coverage_mapping(&mut self, address: u64, output: &StalkerOutput) {
         let h64 = xxh3_rrmxmx_mixer(address);
         let writer = output.writer();
+
+        // Since the AARCH64 instruction set requires that a register be used if
+        // performing a long branch, if the Stalker engine is unable to use a near
+        // branch to transition between branches, then it spills some registers
+        // into the stack beyond the red-zone so that it can use them to perform
+        // the branch. Accordingly each block is transparently prefixed with an
+        // instruction to restore these registers. If however a near branch can
+        // be used, then this instruction is simply skipped and Stalker simply
+        // branches to the second instruction in the block.
+        //
+        // Since we also need to spill some registers in order to update our
+        // coverage map, in the event of a long branch, we can simply re-use
+        // these spilt registers. This, however, means we need to retard the
+        // code writer so that we can overwrite the so-called "restoration
+        // prologue".
+        let pc = writer.pc();
+        writer.reset(pc - 4);
```

Comment:
Perhaps instruction size is not (always) 4 on amd64?

Reply:
My bad. That was obviously only supposed to happen on AARCH64. Let me put together a patch. I can test on Linux x64, but I don't have windows. Are you able to check for me?

Reply:
Yes of course 😉 
... Well in fact the ci will turn green if it works